ESMF/GOCART2G_GridComp/GA_Environment/GA_EnvironmentMod.F90

```diff
@@ -72,6 +72,11 @@ subroutine load_from_config(self, cfg, universal_cfg, rc)
        call ESMF_ConfigGetAttribute (universal_cfg, self%wavelengths_vertint, &
                                      label='wavelengths_for_vertically_integrated_aop_in_nm:', __RC__)
 
+       !   Convert input wavelengths from nm to m for internal use
+
+       self%wavelengths_profile = 1.0e-9 * self%wavelengths_profile
```

Comment:
See previous comment.

Reply:
Again as above, here we are filling `self%wavelengths_vertint` from `wavelengths_for_vertically_integrated_aop_in_nm`, so we convert to meters after that. This one is the `self%wavelengths_vertint` that each "subcomponent" sees. As each one (DU, CA, etc.) calls `load_from_config` to fill their respective states.

@tclune might have more information.